<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A systems biology tool for gene regulatory circuit simulation • sRACIPE</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A systems biology tool for gene regulatory circuit simulation">
<meta property="og:description" content="sRACIPE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sRACIPE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sRACIPE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/articles/simulatingTrajectories.html">Simulating trajectories</a>
    </li>
    <li>
      <a href="../articles/articles/singleModelSimulations.html">Single Model Simulations</a>
    </li>
    <li>
      <a href="../articles/articles/underTheHood.html">Under the hood - design and implementation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vivekkohar/sRACIPE/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A systems biology tool for gene regulatory circuit simulation</h1>
                        <h4 class="author">Vivek Kohar</h4>
            <address class="author_afil">
      The Jackson Laboratory, Bar Harbor, Maine, USA<br><small class="dont-index">Source: <a href="https://github.com/vivekkohar/sRACIPE/blob/master/vignettes/sRACIPE.Rmd"><code>vignettes/sRACIPE.Rmd</code></a></small>
      <div class="hidden name"><code>sRACIPE.Rmd</code></div>

    </address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This document illustrates the use of sRACIPE to simulate any circuit or network or topology (used interchangeably) and analyze the generated data. sRACIPE implements a randomization-based method for gene circuit modeling. It allows us to study the effect of both the gene expression noise and the parametric variation on any gene regulatory circuit (GRC) using only its topology, and simulates an ensemble of models with random kinetic parameters at multiple noise levels. Statistical analysis of the generated gene expressions reveals the basin of attraction and stability of various phenotypic states and their changes associated with intrinsic and extrinsic noises. sRACIPE provides a holistic picture to evaluate the effects of both the stochastic nature of cellular processes and the parametric variation.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<ol style="list-style-type: decimal">
<li>Download the package from Bioconductor.</li>
</ol>
<div class="sourceCode"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"sRACIPE"</span>)</pre></body></html></div>
<p>Or install the development version of the package.</p>
<div class="sourceCode"><html><body><pre class="r">BiocManager::install(“vivekkohar/sRACIPE”)</pre></body></html></div>
<ol start="2" style="list-style-type: decimal">
<li>Load the package into R session.</li>
</ol>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">sRACIPE</span>)</pre></body></html></div>
<pre><code>## Warning: package 'SummarizedExperiment' was built under R version 3.6.1</code></pre>
<pre><code>## Warning: package 'GenomicRanges' was built under R version 3.6.1</code></pre>
<pre><code>## Warning: package 'S4Vectors' was built under R version 3.6.1</code></pre>
<pre><code>## Warning: package 'IRanges' was built under R version 3.6.1</code></pre>
<pre><code>## Warning: package 'BiocParallel' was built under R version 3.6.1</code></pre>
<p>For simplicity, we will start with the toggle switch with mutual inhibition and self activation of both genes as a test case. Additionally, to keep the simulations fast, we will use fewer models, small integration time and longer integration step size. We recommend using the default settings for most parameters for actual simulations.</p>
</div>
</div>
<div id="load-the-circuit" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-circuit" class="anchor"></a>Load the Circuit</h1>
<p>One can use the topology stored in a text file or loaded as a dataframe. The typical format of the topology file is a 3 column file where the first column is name of the source gene, second column is name of the target gene and last column is the interaction type (1 - activation, 2-inhibition). The first line should contain the header (Source Target Interaction). We will work with a demo circuit from the package.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">sRACIPE</span>)

<span class="co"># Load a demo circuit</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"demoCircuit"</span>)
<span class="no">demoCircuit</span></pre></body></html></div>
<pre><code>##   Source Target Type
## 1      A      B    2
## 2      B      A    2
## 3      A      A    1
## 4      B      B    1</code></pre>
</div>
<div id="simulate-the-circuit" class="section level1">
<h1 class="hasAnchor">
<a href="#simulate-the-circuit" class="anchor"></a>Simulate the circuit</h1>
<p>We will use a reduced number of models (using numModels) for demonstration. The simulations will return a RacipeSE object.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeSimulate.html">sracipeSimulate</a></span>(<span class="kw">circuit</span> <span class="kw">=</span> <span class="no">demoCircuit</span>, <span class="kw">numModels</span> <span class="kw">=</span> <span class="fl">20</span>,
                             <span class="kw">plots</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">integrateStepSize</span> <span class="kw">=</span> <span class="fl">0.1</span>,
                             <span class="kw">simulationTime</span> <span class="kw">=</span> <span class="fl">30</span>)</pre></body></html></div>
<pre><code>## circuit file successfully loaded</code></pre>
<pre><code>## Generating gene thresholds</code></pre>
<pre><code>## generating thresholds for uniform distribution1...
## ========================================</code></pre>
</div>
<div id="plotting-the-simulated-data" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-the-simulated-data" class="anchor"></a>Plotting the simulated data</h1>
<p>We can plot the simulated data using the sracipePlotData function or using plots=TRUE in sracipeSimulate. The data can be normalized before plotting. Otherwise it will be normalized by the plotting function. By default, two clusters are identified and models colored according to hierarchical clustering.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeNormalize.html">sracipeNormalize</a></span>(<span class="no">rSet</span>)
<span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipePlotData.html">sracipePlotData</a></span>(<span class="no">rSet</span>, <span class="kw">plotToFile</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p><img src="sRACIPE_files/figure-html/unnamed-chunk-4-1.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-4-2.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-4-3.png" width="672"></p>
</div>
<div id="knockdown-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#knockdown-analysis" class="anchor"></a>Knockdown Analysis</h1>
<p>The simulations can be used to perform in-silico perturbation analysis. For example, here we will limit a gene’s production rate to mimic its knockdown and show how that changes the relative proportion of models in different clusters.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"demoCircuit"</span>)
<span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeSimulate.html">sracipeSimulate</a></span>(<span class="kw">circuit</span> <span class="kw">=</span> <span class="no">demoCircuit</span>,
                             <span class="kw">numModels</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">plots</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                             <span class="kw">integrateStepSize</span> <span class="kw">=</span> <span class="fl">0.1</span>,
                             <span class="kw">simulationTime</span> <span class="kw">=</span> <span class="fl">30</span>)</pre></body></html></div>
<pre><code>## circuit file successfully loaded</code></pre>
<pre><code>## Generating gene thresholds</code></pre>
<pre><code>## generating thresholds for uniform distribution1...
## ========================================</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="no">kd</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeKnockDown.html">sracipeKnockDown</a></span>(<span class="no">rSet</span>, <span class="kw">plotToFile</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                                <span class="kw">reduceProduction</span><span class="kw">=</span><span class="fl">50</span>)</pre></body></html></div>
<p><img src="sRACIPE_files/figure-html/unnamed-chunk-5-1.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-5-2.png" width="672"></p>
</div>
<div id="plot-the-network" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-the-network" class="anchor"></a>Plot the network</h1>
<p>The network can be plotted in an interactive viewer or html file in the results folder.</p>
<div class="sourceCode"><html><body><pre class="r"> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipePlotCircuit.html">sracipePlotCircuit</a></span>(<span class="no">rSet</span>, <span class="kw">plotToFile</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
</div>
<div id="stochastic-simulations" class="section level1">
<h1 class="hasAnchor">
<a href="#stochastic-simulations" class="anchor"></a>Stochastic simulations</h1>
<p>One can perform stochastic simulations similarly by specifying additional parameters to the sracipeSimulate function, namely, nNoise (the number of noise levels at which the stochastic simulations should be carried out), initialNoise (the starting noise level) and noiseScalingFactor (the multiplicative factor by which noise should be reduced for multiple noise levels). For annealing, use anneal=TRUE alongwith the above mentioned parameters. For simulations at one noise level only, use nNoise = 1 and set initialNoise parameter to the specific noise.</p>
<p>Now the returned object will contain additional elements in the assays which correspond to the simulations at different noise levels (noise specified by the name of the element).</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeSimulate.html">sracipeSimulate</a></span>(<span class="kw">circuit</span> <span class="kw">=</span> <span class="no">demoCircuit</span>, <span class="kw">numModels</span> <span class="kw">=</span> <span class="fl">20</span>,
                             <span class="kw">initialNoise</span> <span class="kw">=</span> <span class="fl">15</span>, <span class="kw">noiseScalingFactor</span> <span class="kw">=</span> <span class="fl">0.1</span>,
                             <span class="kw">nNoise</span> <span class="kw">=</span> <span class="fl">2</span>,
                             <span class="kw">plots</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">plotToFile</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                             <span class="kw">integrateStepSize</span> <span class="kw">=</span> <span class="fl">0.1</span>,
                             <span class="kw">simulationTime</span> <span class="kw">=</span> <span class="fl">30</span>)</pre></body></html></div>
<pre><code>## circuit file successfully loaded</code></pre>
<pre><code>## Generating gene thresholds</code></pre>
<pre><code>## generating thresholds for uniform distribution1...
## ========================================</code></pre>
<p><img src="sRACIPE_files/figure-html/unnamed-chunk-7-1.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-7-2.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-7-3.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-7-4.png" width="672"><img src="sRACIPE_files/figure-html/unnamed-chunk-7-5.png" width="672"></p>
<p>Here, calling the sracipeSimulate function simulated the circuit at zero noise level as well as the two (nNoise) other noise levels 15 (initialNoise), 1.5 (initialNoise*noiseScalingFactor). The first three plots (hierarchical clustering heatmap, Umap, PCA) correspond to deterministic data and the last two plots contain the data from stochastic simulations projected on the principal components of the deterministic simulations.</p>
<p>Note that the rSet now contains stochastic simulations as well as additional elements like umap, pca, assignedClusters. These are added when the data is plotted. As mentioned previously, the additional elements of the assays are named “15” and “1.5” which correspond to noise levels.</p>
<p>For annealing simulations, one can set anneal=TRUE in the sracipeSimulate function. With anneal=FALSE (constant noise), simulations at different noise levels are independent of each other. These are useful if one is primarily interested in the gene expressions at different noise levels and at zero noise (used for normalizing the data). With annealing, the steady state solutions at higher noise levels are used as the intial conditions for lower noise levels such that each model converges to its most stable state when the noise is zero.</p>
<p>Using annealing, ideally the number of noise levels should be very large and noiseScalingFactor close to 1 as we want to reduce the noise very slowly. In practice, we found nNoise ~ 30 and initialNoise ~50 divided by sqrt(number_gene) as good starting values. Constant noise and annealing noise simulations pca plots can be used for better approximations of these parameters. The initialNoise should be such that there is a single cluster at this high noise level (essentially the gene expression values are random and circuit topology has little effect). Next, noiseScalingFactor should be adjusted such that there are sufficient noise levels when this single cluster splits into multiple clusters observed in deterministic simulations.</p>
<p>With annealing, the models converge to their most stable steady state at zero noise. Thus, the number of models is more stable clusters will increase and number in less stable clusters will decrease. Note that for non zero noise, the stable states can be different from the stable states at zero noise. In our illustrative example shown abpve, the previous two stable states of a toggle circuit are no longer stable at high noise (“15”) and instead the previously unstable high high state is stable now. Briefly, noise can change the stable states and zero noise simulations using annealing can<br>
gives us an idea about the relative stability of states when there are multiple stable states.</p>
<p>Further, one can modify the parameters and/or initial conditions and simulate the circuit with modified parameters and/or initial conditions using the parameters genParams = FALSE and/or genIC = FALSE.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeSimulate.html">sracipeSimulate</a></span>(<span class="kw">circuit</span> <span class="kw">=</span> <span class="no">demoCircuit</span>, <span class="kw">numModels</span> <span class="kw">=</span> <span class="fl">20</span>,
                             <span class="kw">plots</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">integrate</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<pre><code>## circuit file successfully loaded</code></pre>
<pre><code>## Generating gene thresholds</code></pre>
<pre><code>## generating thresholds for uniform distribution1...
## ========================================</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="no">params</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeParams.html">sracipeParams</a></span>(<span class="no">rSet</span>)
<span class="no">modifiedParams</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">params</span>)
<span class="no">modifiedParams</span>[,<span class="fl">1</span>] <span class="kw">&lt;-</span> <span class="fl">0.1</span>*<span class="no">modifiedParams</span>[,<span class="fl">1</span>]
<span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeParams.html">sracipeParams</a></span>(<span class="no">rSet</span>) <span class="kw">&lt;-</span> <span class="fu">DataFrame</span>(<span class="no">modifiedParams</span>)
<span class="no">rSet</span> <span class="kw">&lt;-</span> <span class="kw pkg">sRACIPE</span><span class="kw ns">::</span><span class="fu"><a href="../reference/sracipeSimulate.html">sracipeSimulate</a></span>(<span class="no">rSet</span>, <span class="kw">plots</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">genParams</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<pre><code>## ========================================</code></pre>
</div>
<div id="knockout-simulations" class="section level1">
<h1 class="hasAnchor">
<a href="#knockout-simulations" class="anchor"></a>KnockOut Simulations</h1>
<p>Knockout of a gene is implemented by changing the production rate and initial condition of the gene to zero. The knockOut parameter in the function sracipeSimulate can be used to perform these knockout simulations. If simulations are to be carried out for knockouts of different genes, the genes should be specified as a list where each list element will contain the names of the gene to be knocked out. For example, knockout = list(“gene1”, “gene2”, c(“gene3”, “gene4”), “gene5”) will knockout gene1, gene2, gene5 one by one and knockout gene3 and gene4 simultaneously. knockOut = “all”, each gene is knocked out one by one and the results are returned as an element knockOutSimulations which, similar to stochasticSimulations, is a list of dataframes containing the gene expressions obtained by knockout one or more genes. Enabling plots=TRUE will plot the results. As the expression of knockout gene is zero, we compute PCA with unperturbed genes for both the unperturbed simulations as well as the perturbed simulations. So for each knockout, we have two plots containing the scatter plot of unperturbed simulations and perturbed simulations on the PCs of unperturbed simulations (excluding the gene to be perturbed).</p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Kohar V, Lu M (2018). “Role of noise and parametric variation in the dynamics of gene regulatory circuits.” npj Systems Biology and Applications, 4, 40. <a href="https://www.nature.com/articles/s41540-018-0076-x" class="uri">https://www.nature.com/articles/s41540-018-0076-x</a>.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] sRACIPE_1.3.4               Rcpp_1.0.4                 
##  [3] SummarizedExperiment_1.14.1 DelayedArray_0.10.0        
##  [5] BiocParallel_1.18.1         matrixStats_0.56.0         
##  [7] Biobase_2.44.0              GenomicRanges_1.36.1       
##  [9] GenomeInfoDb_1.20.0         IRanges_2.18.3             
## [11] S4Vectors_0.22.1            BiocGenerics_0.30.0        
## [13] BiocStyle_2.12.0           
## 
## loaded via a namespace (and not attached):
##  [1] jsonlite_1.6.1         gtools_3.8.2           assertthat_0.2.1      
##  [4] askpass_1.1            BiocManager_1.30.10    GenomeInfoDbData_1.2.1
##  [7] yaml_2.2.1             pillar_1.4.3           backports_1.1.6       
## [10] lattice_0.20-41        glue_1.4.0             reticulate_1.15       
## [13] digest_0.6.25          RColorBrewer_1.1-2     XVector_0.24.0        
## [16] colorspace_1.4-1       htmltools_0.4.0        Matrix_1.2-18         
## [19] plyr_1.8.6             pkgconfig_2.0.3        bookdown_0.18         
## [22] zlibbioc_1.30.0        purrr_0.3.4            scales_1.1.0          
## [25] gdata_2.18.0           RSpectra_0.16-0        tibble_3.0.1          
## [28] openssl_1.4.1          farver_2.0.3           ggplot2_3.3.0         
## [31] ellipsis_0.3.0         umap_0.2.5.0           magrittr_1.5          
## [34] crayon_1.3.4           memoise_1.1.0          evaluate_0.14         
## [37] fs_1.4.1               MASS_7.3-51.5          gplots_3.0.3          
## [40] tools_3.6.0            lifecycle_0.2.0        stringr_1.4.0         
## [43] munsell_0.5.0          compiler_3.6.0         pkgdown_1.5.1         
## [46] caTools_1.18.0         rlang_0.4.5            grid_3.6.0            
## [49] RCurl_1.98-1.2         rstudioapi_0.11        visNetwork_2.0.9      
## [52] htmlwidgets_1.5.1      labeling_0.3           bitops_1.0-6          
## [55] rmarkdown_2.1          gtable_0.3.0           reshape2_1.4.4        
## [58] R6_2.4.1               gridExtra_2.3          knitr_1.28            
## [61] dplyr_0.8.5            rprojroot_1.3-2        KernSmooth_2.23-16    
## [64] desc_1.2.0             stringi_1.4.6          vctrs_0.2.4           
## [67] tidyselect_1.0.0       xfun_0.13</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://vivekkohar.github.io/">Vivek Kohar</a>, <a href="http://lusystemsbio.com/">Mingyang Lu</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
