<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calculates the similarity between two gene expression data. — sracipeHeatmapSimilarity • sRACIPE</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Calculates the similarity between two gene expression data. — sracipeHeatmapSimilarity" />
<meta property="og:description" content="Comparison is done across columns, i.e., 
how similar are the columns in the two dataset.
For gene expression data, format data so that gene names are in rows and 
samples in columns." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sRACIPE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sRACIPE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/articles/simulatingTrajectories.html">Simulating trajectories</a>
    </li>
    <li>
      <a href="../articles/articles/singleModelSimulations.html">Single Model Simulations</a>
    </li>
    <li>
      <a href="../articles/articles/underTheHood.html">Under the hood - design and implementation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/vivekkohar/sRACIPE/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculates the similarity between two gene expression data.</h1>
    <small class="dont-index">Source: <a href='https://github.com/vivekkohar/sRACIPE/blob/master/R/heatmapSimilarity.R'><code>R/heatmapSimilarity.R</code></a></small>
    <div class="hidden name"><code>sracipeHeatmapSimilarity.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Comparison is done across columns, i.e., 
how similar are the columns in the two dataset.
For gene expression data, format data so that gene names are in rows and 
samples in columns.</p>
    </div>

    <pre class="usage"><span class='fu'>sracipeHeatmapSimilarity</span>(
  <span class='no'>dataReference</span>,
  <span class='no'>dataSimulation</span>,
  <span class='kw'>clusterCut</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>nClusters</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>pValue</span> <span class='kw'>=</span> <span class='fl'>0.05</span>,
  <span class='no'>permutedVar</span>,
  <span class='kw'>permutations</span> <span class='kw'>=</span> <span class='fl'>1000</span>,
  <span class='kw'>corMethod</span> <span class='kw'>=</span> <span class='st'>"spearman"</span>,
  <span class='kw'>clusterMethod</span> <span class='kw'>=</span> <span class='st'>"ward.D2"</span>,
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"pvalue"</span>,
  <span class='kw'>buffer</span> <span class='kw'>=</span> <span class='fl'>0.001</span>,
  <span class='kw'>permutMethod</span> <span class='kw'>=</span> <span class='st'>"simulation"</span>,
  <span class='kw'>returnData</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataReference</th>
      <td><p>Matrix. The reference data matrix, for example, 
the experimental gene expression values</p></td>
    </tr>
    <tr>
      <th>dataSimulation</th>
      <td><p>Matrix. The data matrix to be compared.</p></td>
    </tr>
    <tr>
      <th>clusterCut</th>
      <td><p>(optional) Integer vector. Clsuter numbers assigned 
to reference data.
If clusterCut is missing, hierarchical clustering using /codeward.D2
and /codedistance  = (1-cor(x, method = "spear"))/2 will be used to 
cluster the reference data.</p></td>
    </tr>
    <tr>
      <th>nClusters</th>
      <td><p>(optional) Integer. The number of clusters in which the
reference data should be clustered for comparison.
Not needed if clusterCut is provided.</p></td>
    </tr>
    <tr>
      <th>pValue</th>
      <td><p>(optional) Numeric. p-value to consider two gene expression
 sets as belonging to same cluster.
Ward's method with spearman correlation is used to determine if a
model belongs to a specific cluster.</p></td>
    </tr>
    <tr>
      <th>permutedVar</th>
      <td><p>(optional) Similarity scores computed after permutations.</p></td>
    </tr>
    <tr>
      <th>permutations</th>
      <td><p>(optional) Integer. Default <code>1000</code>.
Number of gene permutations to generate the null distibution.</p></td>
    </tr>
    <tr>
      <th>corMethod</th>
      <td><p>(optional) Correlation method. Default method is "spearman".
For single cell data, use "kendall"</p></td>
    </tr>
    <tr>
      <th>clusterMethod</th>
      <td><p>(optional) Character - default <code>ward.D2</code>, other
options include <code>complete</code>. Clustering method to be used to cluster the
 experimental data. <code><a href='https://rdrr.io/r/stats/hclust.html'>hclust</a></code> for other options.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>(optional) character. Method to compare the gene expressions.
Default <code>pvalue</code>. One can use <code>variance</code> as well which assigns
clusters based on the cluster whose samples have minimum variance with
the simulated sample.</p></td>
    </tr>
    <tr>
      <th>buffer</th>
      <td><p>(optional) Numeric. Default <code>0.001</code>. The fraction of
models to be assigned to clusters to which no samples could be assigned.
For example, a minimum of 1 ghost sample in reference is assigned to
 NULL cluster.</p></td>
    </tr>
    <tr>
      <th>permutMethod</th>
      <td><p>"sample" or "reference"</p></td>
    </tr>
    <tr>
      <th>returnData</th>
      <td><p>(optional) Logical. Default <code>FALSE</code>. Whether to
return the sorted and clustered data.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing the KL distance of new cluster distribution from 
reference data and
the probability of each cluster in the reference and simulated data.</p>
    <h2 class="hasAnchor" id="related-functions"><a class="anchor" href="#related-functions"></a>Related Functions</h2>

    


<p><code><a href='sracipeSimulate.html'>sracipeSimulate</a></code>,  <code><a href='sracipeKnockDown.html'>sracipeKnockDown</a></code>,
<code><a href='sracipeOverExp.html'>sracipeOverExp</a></code>,  <code><a href='sracipePlotData.html'>sracipePlotData</a></code>,
<code>sracipeHeatmapSimilarity</code></p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://vivekkohar.github.io/'>Vivek Kohar</a>, <a href='http://lusystemsbio.com/'>Mingyang Lu</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


